ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/notification/Notification.java

```diff
@@ -33,462 +33,99 @@
-     */
-    public Notification(@NotNull String message, @NotNull Type type, boolean important,
-                        @Nullable OpenNotificationHandler openHandler, @Nullable CloseNotificationHandler closeHandler) {
-        this(message, type, FINISHED, UNREAD, new Date(), important, openHandler, closeHandler);
-    }
+public class Notification implements IDProvider {
+
+    protected NotificationListener listener;
+
+    protected String                     id;
+    protected long                       time;
+    private   List<NotificationObserver> observers;
+
+    protected String           title;
+    protected String           message;
+    protected ReadState        state;
+    protected ProjectConfigDto project;
+    protected DelayedTask setUnreadStateTask = new DelayedTask() {
+        @Override
+        public void onExecute() {
+            time = System.currentTimeMillis();
+            setState(UNREAD);
+        }
+    };
 
-    /**
-     * Create notification with message, type, status, note about important this one and action delegates on opening and closing of a
-     * notification. Other parameters will be added with default values. This notification have got an unread state. This notification
-     * delegates some actions in response to opening and closing of a notification.
-     *
-     * @param message
-     *         notification's message
-     * @param type
-     *         notification's type
-     * @param status
-     *         notification's status
-     * @param important
-     *         note about important this notification
-     * @param openHandler
-     *         delegate that provides some actions when opening notification
-     * @param closeHandler
-     *         delegate that provides some actions when closing notification
-     */
-    public Notification(@NotNull String message, @NotNull Type type, @NotNull Status status, boolean important,
-                        @Nullable OpenNotificationHandler openHandler, @Nullable CloseNotificationHandler closeHandler) {
-        this(message, type, status, UNREAD, new Date(), important, openHandler, closeHandler);
+    public Notification(@NotNull String title,
+                        @NotNull String message) {
+        this(null, title, message);
     }
 
-    /**
-     * Create notification with message and status. Other parameters will be added with default values. This notification have
-     * got an unread state, a info type.jr l It will be a non-important message that does not delegate any actions in response to opening
-     * and
-     * closing of a notification.
-     *
-     * @param message
-     *         notification's message
-     * @param status
-     *         notification's status
-     */
-    public Notification(@NotNull String message, @NotNull Status status) {
-        this(message, status, null, null);
+    public Notification(@Nullable ProjectConfigDto project,
+                        @NotNull String title,
```

Comment:
Not every time. Its true for the server side, but for the client side behaviour can be unexpected. So its impossible to define all possible behiours.